1.fee_bumping/fee_bumping.md

```diff
@@ -153,33 +153,224 @@ Bitcoin blockchain, such as exchanges or custodians:
+  time. That could either be because the replacement transaction is directly
+  replacing two or more original tranasactions by spending their inputs, or
+  because the original transaction that is being replaced has descendants in
+  the mempool which will also need to be removed.
+- implication 2: Additional inputs can be added to the replacement transaction.
+  This is useful if a wallet needs to bump the fee on a transaction but there
+  is no change output to subtract the additional fee from.
+- implication 3: The replacement transaction does not need to include all of
+  the inputs from the original transaction. This can be extremely dangerous
+  since wallets can easily overspend this way. For example, if the original
+  transaction uses inputs A and B, a replacement transaction is sent with
+  inputs B and C, and then that replacement transaction is replaced with a
+  new replacement transaction using inputs C and D, the first transaction
+  could be rebroadcast by a third party at any point in the future, and both
+  would be valid! Wallets are advised to include all inputs from the original
+  transaction in the replacement transaction unless they have very reliable
+  tracking to make sure that they don't double-spend themselves.
+- implication 4: the `nSequence` of the inputs in the replacement transaction
+  are not considered, so a replaceable transaction could be replaced by a
+  non-replaceable transaction. If a wallet wants to be able to replace the
+  replacement transaction in future, it must signal opt-in RBF on one of the
+  inputs of the replacement transaction.
+
+> 1. The original transactions signal replaceability explicitly or through
+> inheritance as described in the above Summary section.
+
+- implication 1: It is not sufficient to look at an individual transaction
+  to determine whether it can be replaced under this policy. A transaction
+  may be replaceable because one of its unconfirmed ancestors signals
+  opt-in RBF.
+- implication 2: A transaction's replaceability may change! If it is does
+  not signal opt-in replaceability, but one of its ancestors does, then it
+  is replaceable. If the ancestor then gets confirmed, the descendant
+  transaction becomes non-replaceable.
+
+> 2. The replacement transaction may only include an unconfirmed input if that
+> input was included in one of the original transactions. (An unconfirmed
+> input spends an output from a currently-unconfirmed transaction.)
+
+Rationale: this is an anti-DoS measure to stop a spender creating a replacement
+transaction which has a higher feerate than the original transaction, but
+adds new ancestor dependencies so that the ancestor feerate for the transaction
+is lower.
+
+- implication 1: if a wallet runs out of confirmed transactions (ie all of its
+  coins are tied up in unconfirmed transactions), then it can't use the output
+  of some of those transactions as additional inputs to bump the fee on its other
+  transactions.
```

Comment:
I think the conditional clause that opens this example confuses it by making it sound like the ability to use unconfirmed inputs is conditional on whether a confirmed output is also included.  E.g. it sounds like Alice could include an unconfirmed output into an RBF if she also includes a new confirmed output at the same time.

I suggest changing this to something like, "Wallets can't use the outputs of their unconfirmed transactions as new inputs to their RBFs.  If a wallet runs out of confirmed transactions, it can only RBF a transaction by decreasing the value of at least one of the outputs in that transaction."